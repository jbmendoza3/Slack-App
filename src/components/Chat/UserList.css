.user-list-modal {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background-color: transparent;
    display: flex;
    justify-content: center;
    align-items: center;
    color: whitesmoke;
  }
  
  .modal-content {
    background-color: rgb(51, 51, 51);
    padding: 20px;
    border-radius: 8px;
    width: 300px;
    text-align: center;
    font-size: 20px;
  }
  
  ul {
    list-style-type: none;
    padding: 0;
  }
  
  li {
    margin: 10px 0;
  }
  