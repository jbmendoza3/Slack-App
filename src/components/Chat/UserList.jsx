import React, { useEffect, useState } from 'react';
import UserService from '../../services/UserService';
import './UserList.css';

const UserList = ({ channelId, onClose }) => {
  const [users, setUsers] = useState([]);
  const user = JSON.parse(localStorage.getItem('user'));

  useEffect(() => {
    const fetchUsers = async () => {
        if(users.length <= 0) {
            try {
                const usersInChannel = await UserService.getUsersInChannel(user, channelId);
                setUsers(usersInChannel);
              } catch (error) {
                alert('Error fetching user list');
            }
        }
        console.log(users)
    };

    fetchUsers();
  }, [channelId, user, users]);

  return (
    <div className="user-list-modal">
      <div className="modal-content">
        <h2>Channel Users</h2>
        <ul>
          {users && users.map((person) => (
            <li key={person["user_id"]}>({person["user_id"]})</li>
          ))}
        </ul>
        <button onClick={onClose}>Close</button>
      </div>
    </div>
  );
};

export default UserList;
