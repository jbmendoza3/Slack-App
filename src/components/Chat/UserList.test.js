import { render, screen } from '@testing-library/react'
import UserList from './UserList'
import renderer from 'react-test-renderer'

describe('UserList', () => {
    it('should render UserList', () => {
        render(<UserList />);

        expect(screen.getByText(/Channel Users/i)).toBeInTheDocument;
    });
});

test('snapshot test for userlist', () => {
    const userlist = renderer.create(<UserList />).toJSON();
    expect(userlist).toMatchSnapshot();
})