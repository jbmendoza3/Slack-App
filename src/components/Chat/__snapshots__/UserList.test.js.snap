// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for userlist 1`] = `
<div
  className="user-list-modal"
>
  <div
    className="modal-content"
  >
    <h2>
      Channel Users
    </h2>
    <ul />
    <button>
      Close
    </button>
  </div>
</div>
`;
